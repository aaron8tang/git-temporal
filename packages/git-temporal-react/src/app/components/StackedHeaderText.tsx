import React from 'react';

import { StackedLabelText, StackedLabelTextProps } from './StackedLabelText';

const localStyles = {
  labelStyle: {
    _extends: 'h3Text',
    marginBottom: 5,
  },
  textStyle: {
    _extends: 'h2Text',
  },
};

export const StackedHeaderText = (
  props: StackedLabelTextProps
): JSX.Element => {
  const propsForward = Object.assign({}, props, {
    labelStyle: [localStyles.labelStyle, props.labelStyle],
    textStyle: [localStyles.textStyle, props.textStyle],
  });

  return <StackedLabelText {...propsForward} />;
};
