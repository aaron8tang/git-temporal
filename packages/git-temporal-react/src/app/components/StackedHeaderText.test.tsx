import React from 'react';
import { shallow } from 'enzyme';

import { StackedHeaderText } from './StackedHeaderText';

describe('components/StackedHeaderText', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <StackedHeaderText label="This is the Label" text="this is the text" />
      );
    });

    test('it should match snapshot (it should have rendered a label and a text element)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
