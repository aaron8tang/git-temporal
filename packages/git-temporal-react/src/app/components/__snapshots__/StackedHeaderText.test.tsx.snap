// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StackedHeaderText when rendered with required props it should match snapshot (it should have rendered a label and a text element) 1`] = `
<Component
  label="This is the Label"
  labelStyle={
    Array [
      Object {
        "_extends": "h3Text",
        "marginBottom": 5,
      },
      undefined,
    ]
  }
  text="this is the text"
  textStyle={
    Array [
      Object {
        "_extends": "h2Text",
      },
      undefined,
    ]
  }
/>
`;
